apps/examples/testcase/le_tc/kernel/tc_libc_spawn.c

```diff
@@ -381,17 +435,36 @@ static void tc_libc_spawn_posix_spawn_file_actions_dump(void)
 	ret_chk = posix_spawn_file_actions_init(&st_fileactions);
 	TC_ASSERT_EQ("posix_spawn_file_actions_init", ret_chk, OK);
 
+	posix_spawn_file_actions_dump(&st_fileactions);
+
 	ret_chk = posix_spawn_file_actions_addopen(&st_fileactions, 1, szfilepath, O_WRONLY, 0644);
-	TC_ASSERT_EQ("posix_spawn_file_actions_addopen", ret_chk, OK);
+	TC_ASSERT_EQ_CLEANUP("posix_spawn_file_actions_addopen", ret_chk, OK, goto errout);
 
 	/* posix_spawn_file_actions_dump returns (void) */
 
 	posix_spawn_file_actions_dump(&st_fileactions);
 
+	ret_chk = posix_spawn_file_actions_adddup2(&st_fileactions, 1, 2);
+	TC_ASSERT_EQ_CLEANUP("posix_spawn_file_actions_adddup2", ret_chk, OK, goto errout);
+
+	posix_spawn_file_actions_dump(&st_fileactions);
+
+	ret_chk = posix_spawn_file_actions_addclose(&st_fileactions, 1);
+	TC_ASSERT_EQ_CLEANUP("posix_spawn_file_actions_addclose", ret_chk, OK, goto errout);
+
+	posix_spawn_file_actions_dump(&st_fileactions);
+
+	/* Destroy each file action, one at a time */
+
 	ret_chk = posix_spawn_file_actions_destroy(&st_fileactions);
 	TC_ASSERT_EQ("posix_spawn_file_actions_destroy", ret_chk, OK);
 
 	TC_SUCCESS_RESULT();
+
+errout:
+	ret_chk = posix_spawn_file_actions_destroy(&st_fileactions);
+	TC_ASSERT_EQ("posix_spawn_file_actions_destroy", ret_chk, OK);
```

Comment:
errout means this tc already failed. Do we need to use TC_ASSERT macro in here again? What's the meaning of success and failure here?

Reply:
it is API specific (_destroy failure). 
We can skip this TC_ASSERT if not required.

Reply:
I think we should remove this line, because we are already in failure case.